lib/charms/nrpe_exporter/v0/nrpe_exporter.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2021 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""## Overview.
+
+## NRPE Exporter Library Usage
+
+The `NrpeExporterProvider` object may be used by charms to call
+configured `nrpe` daemons and export the job results to Prometheus
+metrics to facilitate migrating from Nagios+NRPE to
+Prometheus+Alertmanager. In order to do this, a charm must
+instantiate the `NrpeExporterProvider` object, and optionally a list
+of relations which will be checked for `monitor` data to configure
+Prometheus jobs.
+
+The parent charm should also provide an instantiated
+`MetricsEndpointAggregator`, so events sent from this library can be
+consumed and sent through the Prometheus relation, by responding to
+`NrpeTargetsChangedEvent`.
+
+
+    self.framework.observe(
+        self.nrpe_exporter_provider.on.nrpe_targets_changed,
+        self._on_nrpe_targets_changed,
+    )
+"""
```

Comment:
It was difficult for me to follow the overview but maybe it's just because I'm not familiar.